gluten-substrait/src/main/scala/org/apache/gluten/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -160,7 +178,7 @@ object FilterExecTransformerBase {
   }
 }
 
-case class ProjectExecTransformer private (projectList: Seq[NamedExpression], child: SparkPlan)
+abstract class ProjectExecTransformerBase(val list: Seq[NamedExpression], val input: SparkPlan)
```

Comment:
@zhztheplayer here is a change from case class to abstract class, any concerns about this?